Documentation/dev-guide/api_grpc_gateway.md

```diff
@@ -0,0 +1,38 @@
+
+## Why grpc-gateway
+
+etcd v3 uses [gRPC][grpc]. Most recommended is to use etcd [Go client][go-client] and [etcdctl][etcdctl]. However, some languages are not supported by gRPC. This is where [grpc-gateway][grpc-gateway] steps in. It builds a RESTful proxy based on the gRPC services, which can serve HTTP and JSON requests.
+
+
+## How grpc-gateway
+
+gRPC messages are in Protocol Buffer, serialized binary data. So requests to grpc-gateway need to be base64-encoded.
```

Comment:
Note that key, value are defined as bytes in Protocol Buffer, thus need to be encoded as base64 string.